From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Thu, 24 Apr 2025 17:39:03 -0500
Subject: [PATCH] Jellyfish Command And Config


diff --git a/net/minecraft/server/Main.java b/net/minecraft/server/Main.java
index bc66a97c59759b08c471773ac705b2e8f442796f..ac9a41b8ca7d89220627a9832e65b574be83d942 100644
--- a/net/minecraft/server/Main.java
+++ b/net/minecraft/server/Main.java
@@ -109,6 +109,7 @@ public class Main {
                 JvmProfiler.INSTANCE.start(Environment.SERVER);
             }
 
+            new net.serlith.jellyfish.JellyfishConfig().load(); // Jellyfish - Jellyfish config
             io.papermc.paper.plugin.PluginInitializerManager.load(optionSet); // Paper
             Bootstrap.bootStrap();
             Bootstrap.validate();
diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 91403df719574d668e6f4f2165a3925a41fe9c86..c9b71e761e9eb662191a2e1efcfa63b7902ada37 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -186,6 +186,11 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         com.destroystokyo.paper.VersionHistoryManager.INSTANCE.getClass(); // Paper - load version history now
         new gg.pufferfish.pufferfish.PufferfishConfig().load(); // Pufferfish - Config and command
 
+        // Jellyfish start - Jellyfish config and command
+        net.serlith.jellyfish.JellyfishConfig.init();
+        net.serlith.jellyfish.JellyfishCommand.init();
+        // Jellyfish end - Jellyfish config and command
+
         this.setPvpAllowed(properties.pvp);
         this.setFlightAllowed(properties.allowFlight);
         this.setMotd(properties.motd);
