From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Wed, 19 Mar 2025 01:18:39 -0500
Subject: [PATCH] Pufferfish Check for spooky season


diff --git a/net/minecraft/world/entity/ambient/Bat.java b/net/minecraft/world/entity/ambient/Bat.java
index 5ebe7b1dce367d5c5e1136b97b2b9f6737595201..8b6d9366ee1a046de5256f4167710a9bc5b251cc 100644
--- a/net/minecraft/world/entity/ambient/Bat.java
+++ b/net/minecraft/world/entity/ambient/Bat.java
@@ -243,12 +243,21 @@ public class Bat extends AmbientCreature {
         }
     }
 
+    // Pufferfish start - only check for spooky season once an hour
+    private static boolean isSpookySeason = false;
+    private static final int ONE_HOUR = 20 * 60 * 60;
+    private static int lastSpookyCheck = -ONE_HOUR;
     private static boolean isHalloween() {
-        LocalDate localDate = LocalDate.now();
-        int i = localDate.get(ChronoField.DAY_OF_MONTH);
-        int i1 = localDate.get(ChronoField.MONTH_OF_YEAR);
-        return i1 == 10 && i >= 20 || i1 == 11 && i <= 3;
+        if (net.minecraft.server.MinecraftServer.currentTick - lastSpookyCheck > ONE_HOUR) {
+            LocalDate localDate = LocalDate.now();
+            int i = localDate.get(ChronoField.DAY_OF_MONTH);
+            int i1 = localDate.get(ChronoField.MONTH_OF_YEAR);
+            isSpookySeason = i1 == 10 && i >= 20 || i1 == 11 && i <= 3;
+            lastSpookyCheck = net.minecraft.server.MinecraftServer.currentTick;
+        }
+        return isSpookySeason;
     }
+    // Pufferfish end - only check for spooky season once an hour
 
     private void setupAnimationStates() {
         if (this.isResting()) {
