From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Thu, 24 Apr 2025 19:56:58 -0500
Subject: [PATCH] Jellyfish No Mob Spawning


diff --git a/net/minecraft/server/level/ServerChunkCache.java b/net/minecraft/server/level/ServerChunkCache.java
index 69d7e2422b58613030a9c5c5af442be25af7e845..b6a74dc923dfa0a3d758c7b6b401d362849ce94e 100644
--- a/net/minecraft/server/level/ServerChunkCache.java
+++ b/net/minecraft/server/level/ServerChunkCache.java
@@ -524,6 +524,7 @@ public class ServerChunkCache extends ChunkSource implements ca.spottedleaf.moon
             this.broadcastChangedChunks(profilerFiller);
             profilerFiller.pop();
         }
+        /* Jellyfish start - No mob spawning
         // Pufferfish start - Optimize mob spawning
         if (gg.pufferfish.pufferfish.PufferfishConfig._ENABLE_ASYNC_MOB_SPAWNING) {
             for (ServerPlayer player : this.level.players) {
@@ -560,6 +561,7 @@ public class ServerChunkCache extends ChunkSource implements ca.spottedleaf.moon
             }
         }
         // Pufferfish end - Optimize mob spawning
+        Jellyfish end - No mob spawning */
     }
 
     private void broadcastChangedChunks(ProfilerFiller profiler) {
@@ -577,6 +579,7 @@ public class ServerChunkCache extends ChunkSource implements ca.spottedleaf.moon
     }
 
     private void tickChunks(ProfilerFiller profiler, long timeInhabited) {
+        /* Jellyfish start - No mob spawning
         profiler.popPush("naturalSpawnCount");
         int naturalSpawnChunkCount = this.distanceManager.getNaturalSpawnChunkCount();
         // Paper start - Optional per player mob spawns
@@ -652,6 +655,7 @@ public class ServerChunkCache extends ChunkSource implements ca.spottedleaf.moon
         if (_boolean) {
             this.level.tickCustomSpawners(this.spawnEnemies, this.spawnFriendlies);
         }
+        Jellyfish start - No mob spawning */
     }
 
     private void tickSpawningChunk(LevelChunk chunk, long timeInhabited, List<MobCategory> spawnCategories, NaturalSpawner.SpawnState spawnState) {
