From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Thu, 20 Mar 2025 00:17:13 -0500
Subject: [PATCH] Toggle Mob AI


diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index 451b3c9a3cde8d26d3f7be40e0d64240e1260f01..a31e9dfefa0cc94980a9be076b8f78561bb802c4 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -3157,7 +3157,7 @@ public abstract class LivingEntity extends Entity implements Attackable {
             }
         }
 
-        if (!this.isRemoved()) {
+        if (!net.serlith.jellyfish.JellyfishConfig.noTickMobAI && !this.isRemoved()) { // Jellyfish - Don't tick mob AI
             this.aiStep();
         }
 
