From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Thu, 20 Mar 2025 01:25:40 -0500
Subject: [PATCH] Player Below World Event


diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index a31e9dfefa0cc94980a9be076b8f78561bb802c4..1a5523f7b4ab71d0e29ae6e36dc9aaab4c5f63a4 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -2607,6 +2607,11 @@ public abstract class LivingEntity extends Entity implements Attackable {
     @Override
     protected void onBelowWorld() {
         this.hurt(this.damageSources().fellOutOfWorld(), this.level().getWorld().getVoidDamageAmount()); // Paper - use configured void damage amount
+        // Jellyfish start - call PlayerBelowWorldEvent
+        if (!net.serlith.jellyfish.JellyfishConfig.disablePlayerBelowWorldEvent && (this instanceof ServerPlayer player)) {
+            new net.serlith.jellyfish.event.player.PlayerBelowWorldEvent(player.getBukkitEntity()).callEvent();
+        }
+        // Jellyfish end - call PlayerBelowWorldEvent
     }
 
     protected void updateSwingTime() {
