From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Mon, 28 Apr 2025 20:55:56 -0500
Subject: [PATCH] Jellyfish Toggle Tile Entity Ticking


diff --git a/net/minecraft/world/level/block/entity/BlockEntityType.java b/net/minecraft/world/level/block/entity/BlockEntityType.java
index 386e6a48701b4c9256e33174123381a93d61e292..abc919c01362bdb6d42f8ad4096b59e63001913d 100644
--- a/net/minecraft/world/level/block/entity/BlockEntityType.java
+++ b/net/minecraft/world/level/block/entity/BlockEntityType.java
@@ -236,12 +236,19 @@ public class BlockEntityType<T extends BlockEntity> {
     private final BlockEntityType.BlockEntitySupplier<? extends T> factory;
     public final Set<Block> validBlocks;
     private final Holder.Reference<BlockEntityType<?>> builtInRegistryHolder = BuiltInRegistries.BLOCK_ENTITY_TYPE.createIntrusiveHolder(this);
+    public boolean noTick = true; // Jellyfish - Toggle block entity ticking
 
     @Nullable
     public static ResourceLocation getKey(BlockEntityType<?> blockEntityType) {
         return BuiltInRegistries.BLOCK_ENTITY_TYPE.getKey(blockEntityType);
     }
 
+    // Jellyfish start - Toggle block entity ticking
+    public static java.util.Optional<BlockEntityType<?>> byString(String key) {
+        return BuiltInRegistries.BLOCK_ENTITY_TYPE.getOptional(ResourceLocation.tryParse(key));
+    }
+    // Jellyfish end - Toggle block entity ticking
+
     private static <T extends BlockEntity> BlockEntityType<T> register(
         String name, BlockEntityType.BlockEntitySupplier<? extends T> factory, Block... validBlocks
     ) {
diff --git a/net/minecraft/world/level/chunk/LevelChunk.java b/net/minecraft/world/level/chunk/LevelChunk.java
index 14c332cbd65c6286c8e6ed2a4ba62f9b3a3a55c5..62430e62311b93c307b48d8794c0c0c69679b20b 100644
--- a/net/minecraft/world/level/chunk/LevelChunk.java
+++ b/net/minecraft/world/level/chunk/LevelChunk.java
@@ -888,6 +888,7 @@ public class LevelChunk extends ChunkAccess implements ca.spottedleaf.moonrise.p
     }
 
     private <T extends BlockEntity> void updateBlockEntityTicker(T blockEntity) {
+        if (blockEntity.getType().noTick) return; // Jellyfish - Toggle block entity ticking
         BlockState blockState = blockEntity.getBlockState();
         BlockEntityTicker<T> ticker = blockState.getTicker(this.level, (BlockEntityType<T>)blockEntity.getType());
         if (ticker == null) {
