From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Thu, 24 Apr 2025 17:39:03 -0500
Subject: [PATCH] Jellyfish Command And Config


diff --git a/net/minecraft/server/Main.java b/net/minecraft/server/Main.java
index 26b82449334ec6eebc256cd2b9b4be565e8a5264..5a197b22f0fc44c97e589a29baeb4de56cc2ac89 100644
--- a/net/minecraft/server/Main.java
+++ b/net/minecraft/server/Main.java
@@ -108,6 +108,7 @@ public class Main {
                 JvmProfiler.INSTANCE.start(Environment.SERVER);
             }
 
+            new net.serlith.jellyfish.JellyfishConfig().load(); // Jellyfish - Jellyfish config
             io.papermc.paper.plugin.PluginInitializerManager.load(optionSet); // Paper
             Bootstrap.bootStrap();
             Bootstrap.validate();
diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 57f2e627ea449fb0ce1ab41cfce6854babc94d63..83a9e1eb842a955bd10475cbb19b4d613b7a56b9 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -190,6 +190,11 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         gg.pufferfish.pufferfish.PufferfishCommand.init();
         // Pufferfish end - Config and command
 
+        // Jellyfish start - Jellyfish config and command
+        net.serlith.jellyfish.JellyfishConfig.init();
+        net.serlith.jellyfish.JellyfishCommand.init();
+        // Jellyfish end - Jellyfish config and command
+
         this.setPvpAllowed(properties.pvp);
         this.setFlightAllowed(properties.allowFlight);
         this.setMotd(properties.motd);
