From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Thu, 24 Apr 2025 19:04:11 -0500
Subject: [PATCH] Jellyfish Configurable Dimensions


diff --git a/net/minecraft/data/worldgen/DimensionTypes.java b/net/minecraft/data/worldgen/DimensionTypes.java
index 1356ae93cca0a9e16112f61b599eec7d2ea7efd7..8b751b21443f94095855515766eb4894cd9a1acb 100644
--- a/net/minecraft/data/worldgen/DimensionTypes.java
+++ b/net/minecraft/data/worldgen/DimensionTypes.java
@@ -15,18 +15,18 @@ public class DimensionTypes {
                 OptionalLong.empty(),
                 true,
                 false,
-                false,
+                !net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD.PLACE_WATER, // Jellyfish - Dimensions configurable
                 true,
                 1.0,
-                true,
-                false,
-                -64,
-                384,
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD.BED_WORKS, // Jellyfish - Dimensions configurable
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD.RESPAWN_ANCHOR_WORKS, // Jellyfish - Dimensions configurable
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD.MIN_HEIGHT, // Jellyfish - Dimensions configurable
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD.MAX_HEIGHT, // Jellyfish - Dimensions configurable
                 384,
                 BlockTags.INFINIBURN_OVERWORLD,
                 BuiltinDimensionTypes.OVERWORLD_EFFECTS,
-                0.0F,
-                new DimensionType.MonsterSettings(false, true, UniformInt.of(0, 7), 0)
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD.AMBIENT_LIGHT, // Jellyfish - Dimensions configurable
+                new DimensionType.MonsterSettings(net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD.PIGLIN_SAFE, false, UniformInt.of(0, 7), 0) // Jellyfish - Dimensions configurable
             )
         );
         context.register(
@@ -35,18 +35,18 @@ public class DimensionTypes {
                 OptionalLong.of(18000L),
                 false,
                 true,
-                true,
+                !net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.NETHER.PLACE_WATER, // Jellyfish - Dimensions configurable
                 false,
                 8.0,
-                false,
-                true,
-                0,
-                256,
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.NETHER.BED_WORKS, // Jellyfish - Dimensions configurable
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.NETHER.RESPAWN_ANCHOR_WORKS, // Jellyfish - Dimensions configurable
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.NETHER.MIN_HEIGHT, // Jellyfish - Dimensions configurable
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.NETHER.MAX_HEIGHT, // Jellyfish - Dimensions configurable
                 128,
                 BlockTags.INFINIBURN_NETHER,
                 BuiltinDimensionTypes.NETHER_EFFECTS,
-                0.1F,
-                new DimensionType.MonsterSettings(true, false, ConstantInt.of(7), 15)
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.NETHER.AMBIENT_LIGHT, // Jellyfish - Dimensions configurable
+                new DimensionType.MonsterSettings(net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.NETHER.PIGLIN_SAFE, false, ConstantInt.of(7), 15) // Jellyfish - Dimensions configurable
             )
         );
         context.register(
@@ -55,18 +55,18 @@ public class DimensionTypes {
                 OptionalLong.of(6000L),
                 false,
                 false,
-                false,
+                !net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.THE_END.PLACE_WATER, // Jellyfish - Dimensions configurable
                 false,
                 1.0,
-                false,
-                false,
-                0,
-                256,
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.THE_END.BED_WORKS, // Jellyfish - Dimensions configurable
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.THE_END.RESPAWN_ANCHOR_WORKS, // Jellyfish - Dimensions configurable
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.THE_END.MIN_HEIGHT, // Jellyfish - Dimensions configurable
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.THE_END.MAX_HEIGHT, // Jellyfish - Dimensions configurable
                 256,
                 BlockTags.INFINIBURN_END,
                 BuiltinDimensionTypes.END_EFFECTS,
-                0.0F,
-                new DimensionType.MonsterSettings(false, true, UniformInt.of(0, 7), 0)
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.THE_END.AMBIENT_LIGHT, // Jellyfish - Dimensions configurable
+                new DimensionType.MonsterSettings(net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.THE_END.PIGLIN_SAFE, false, UniformInt.of(0, 7), 0) // Jellyfish - Dimensions configurable
             )
         );
         context.register(
@@ -75,18 +75,18 @@ public class DimensionTypes {
                 OptionalLong.empty(),
                 true,
                 true,
-                false,
+                !net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD_CAVES.PLACE_WATER, // Jellyfish - Dimensions configurable
                 true,
                 1.0,
-                true,
-                false,
-                -64,
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD_CAVES.BED_WORKS, // Jellyfish - Dimensions configurable
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD_CAVES.RESPAWN_ANCHOR_WORKS, // Jellyfish - Dimensions configurable
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD_CAVES.MIN_HEIGHT, // Jellyfish - Dimensions configurable
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD_CAVES.MAX_HEIGHT, // Jellyfish - Dimensions configurable
                 384,
-                384,
-                BlockTags.INFINIBURN_OVERWORLD,
+                BlockTags.INFINIBURN_OVERWORLD, // Jellyfish - Dimensions configurable
                 BuiltinDimensionTypes.OVERWORLD_EFFECTS,
-                0.0F,
-                new DimensionType.MonsterSettings(false, true, UniformInt.of(0, 7), 0)
+                net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD_CAVES.AMBIENT_LIGHT,
+                new DimensionType.MonsterSettings(net.serlith.jellyfish.JellyfishConfig.DIMENSIONS.OVERWORLD_CAVES.PIGLIN_SAFE, false, UniformInt.of(0, 7), 0) // Jellyfish - Dimensions configurable
             )
         );
     }
