From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Thu, 24 Apr 2025 19:48:27 -0500
Subject: [PATCH] Jellyfish No Tick Entities


diff --git a/io/papermc/paper/entity/activation/ActivationRange.java b/io/papermc/paper/entity/activation/ActivationRange.java
index 2fa9b8a33e9fb617322d125f9958feb2fcd5edb0..abae01336bf90037060e7c7407cb58f5a26b89af 100644
--- a/io/papermc/paper/entity/activation/ActivationRange.java
+++ b/io/papermc/paper/entity/activation/ActivationRange.java
@@ -300,6 +300,7 @@ public final class ActivationRange {
      * @return
      */
     public static boolean checkIfActive(final Entity entity) {
+        if (entity.getType().noTick) return false; // Jellyfish - Do not tick entities
         // Never safe to skip fireworks or item gravity
         if (entity instanceof FireworkRocketEntity || (entity instanceof ItemEntity && (entity.tickCount + entity.getId()) % 4 == 0)) { // Needed for item gravity, see ItemEntity tick
             return true;
diff --git a/net/minecraft/world/entity/EntityType.java b/net/minecraft/world/entity/EntityType.java
index eb87ab42b4c9f3f0cc1a646ca289b2cba05b430a..593fd178b9f0d8e94c80ba005bc5fc6d092e8cc1 100644
--- a/net/minecraft/world/entity/EntityType.java
+++ b/net/minecraft/world/entity/EntityType.java
@@ -1077,6 +1077,7 @@ public class EntityType<T extends Entity> implements FeatureElement, EntityTypeT
     private final int updateInterval;
     private final String descriptionId;
     public boolean dabEnabled = false; // Pufferfish - DAB
+    public boolean noTick = false; // Jellyfish - No tick entities
     @Nullable
     private Component description;
     private final Optional<ResourceKey<LootTable>> lootTable;
