From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Mon, 28 Apr 2025 15:09:27 -0500
Subject: [PATCH] Leaf Cache Player Profiles


diff --git a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
index 7950f4f88d8a83ed5610b7af4e134557d32da3f0..9a868ceaee2a70ef4bae28ff4245f5aabcc3c8f9 100644
--- a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
@@ -70,6 +70,11 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
     private net.minecraft.server.level.ServerPlayer player; // CraftBukkit
     public boolean iKnowThisMayNotBeTheBestIdeaButPleaseDisableUsernameValidation = false; // Paper - username validation overriding
     private int velocityLoginMessageId = -1; // Paper - Add Velocity IP Forwarding Support
+    // Leaf start - Cache player profileResult
+    private static final com.github.benmanes.caffeine.cache.Cache<String, ProfileResult> playerProfileResultCache = com.github.benmanes.caffeine.cache.Caffeine.newBuilder()
+        .expireAfterWrite(net.serlith.jellyfish.JellyfishConfig.CACHES.PLAYER_PROFILE.TIMEOUT, java.util.concurrent.TimeUnit.MINUTES)
+        .build();
+    // Leaf end - Cache player profileResult
 
     public ServerLoginPacketListenerImpl(MinecraftServer server, Connection connection, boolean transferred) {
         this.server = server;
@@ -289,9 +294,20 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
                 String string1 = Objects.requireNonNull(ServerLoginPacketListenerImpl.this.requestedUsername, "Player name not initialized");
 
                 try {
-                    ProfileResult profileResult = ServerLoginPacketListenerImpl.this.server
-                        .getSessionService()
-                        .hasJoinedServer(string1, string, this.getAddress());
+                    ProfileResult profileResult;
+                    if (net.serlith.jellyfish.JellyfishConfig.CACHES.PLAYER_PROFILE.ENABLED) {
+                        profileResult = playerProfileResultCache.getIfPresent(string1);
+                        if (profileResult == null) {
+                            profileResult = ServerLoginPacketListenerImpl.this.server
+                                .getSessionService()
+                                .hasJoinedServer(string1, string, this.getAddress());
+                            if (profileResult != null) playerProfileResultCache.put(string1, profileResult);
+                        }
+                    } else {
+                        profileResult = ServerLoginPacketListenerImpl.this.server
+                            .getSessionService()
+                            .hasJoinedServer(string1, string, this.getAddress());
+                    }
                     if (profileResult != null) {
                         GameProfile gameProfile = profileResult.profile();
                         // CraftBukkit start - fire PlayerPreLoginEvent
