--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -323,6 +_,13 @@
         // Ensure that player inventory is populated with its viewer
         player.containerMenu.transferTo(player.containerMenu, bukkitPlayer);
 
+        // Jellyfish start - Enable flight during development
+        if (net.serlith.jellyfish.JellyfishConfig.flyOnJoin) {
+            player.getAbilities().flying = true;
+            player.onUpdateAbilities();
+        }
+        // Jellyfish end - Enable flight during development
+
         org.bukkit.event.player.PlayerJoinEvent playerJoinEvent = new org.bukkit.event.player.PlayerJoinEvent(bukkitPlayer, io.papermc.paper.adventure.PaperAdventure.asAdventure(mutableComponent)); // Paper - Adventure
         this.cserver.getPluginManager().callEvent(playerJoinEvent);
 
