From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <contact@serlith.net>
Date: Sat, 8 Mar 2025 01:06:51 -0500
Subject: [PATCH] Builtin Kotlin Library Loader


diff --git a/src/main/java/io/papermc/paper/plugin/provider/configuration/PaperPluginMeta.java b/src/main/java/io/papermc/paper/plugin/provider/configuration/PaperPluginMeta.java
index d3b3a8baca013909fa9c6204d964d7d7efeb2719..e8035bf9d881f45d2adfc9b43675b6b3a353cf12 100644
--- a/src/main/java/io/papermc/paper/plugin/provider/configuration/PaperPluginMeta.java
+++ b/src/main/java/io/papermc/paper/plugin/provider/configuration/PaperPluginMeta.java
@@ -50,6 +50,7 @@ public class PaperPluginMeta implements PluginMeta {
     private String bootstrapper;
     @PluginConfigConstraints.PluginNameSpace
     private String loader;
+    private String kotlin; // Jellyfish - Kotlin library downloader
     private List<String> provides = List.of();
     private boolean hasOpenClassloader = false;
     @Required
@@ -264,6 +265,12 @@ public class PaperPluginMeta implements PluginMeta {
         return this.loader;
     }
 
+    // Jellyfish start - Kotlin library downloader
+    public String getKotlin() {
+        return this.kotlin;
+    }
+    // Jellyfish end - Kotlin library downloader
+
     public boolean hasOpenClassloader() {
         return this.hasOpenClassloader;
     }
diff --git a/src/main/java/io/papermc/paper/plugin/provider/type/paper/PaperPluginProviderFactory.java b/src/main/java/io/papermc/paper/plugin/provider/type/paper/PaperPluginProviderFactory.java
index 0a27b468560ccf4b9588cd12d50c02e442f3024f..d610f22811021d8ba9606233e7bd6f2e1049a33a 100644
--- a/src/main/java/io/papermc/paper/plugin/provider/type/paper/PaperPluginProviderFactory.java
+++ b/src/main/java/io/papermc/paper/plugin/provider/type/paper/PaperPluginProviderFactory.java
@@ -7,6 +7,7 @@ import io.papermc.paper.plugin.entrypoint.classloader.PaperPluginClassLoader;
 import io.papermc.paper.plugin.entrypoint.classloader.PaperSimplePluginClassLoader;
 import io.papermc.paper.plugin.loader.PaperClasspathBuilder;
 import io.papermc.paper.plugin.loader.PluginLoader;
+import io.papermc.paper.plugin.loader.library.impl.MavenLibraryResolver;
 import io.papermc.paper.plugin.provider.configuration.PaperPluginMeta;
 import io.papermc.paper.plugin.provider.type.PluginTypeFactory;
 import io.papermc.paper.plugin.provider.util.ProviderUtil;
@@ -19,6 +20,9 @@ import java.util.jar.JarEntry;
 import java.util.jar.JarFile;
 import java.util.logging.Logger;
 import net.kyori.adventure.text.logger.slf4j.ComponentLogger;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.repository.RemoteRepository;
 
 class PaperPluginProviderFactory implements PluginTypeFactory<PaperPluginParent, PaperPluginMeta> {
 
@@ -41,6 +45,16 @@ class PaperPluginProviderFactory implements PluginTypeFactory<PaperPluginParent,
             }
         }
 
+        // Jellyfish start - Kotlin library downloader
+        if (configuration.getKotlin() != null) {
+            MavenLibraryResolver resolver = new MavenLibraryResolver();
+            resolver.addRepository(new RemoteRepository.Builder("central", "default", "https://repo.maven.apache.org/maven2/").build());
+            resolver.addDependency(new Dependency(new DefaultArtifact("org.jetbrains.kotlin:kotlin-stdlib:" + configuration.getKotlin()), null));
+            resolver.addDependency(new Dependency(new DefaultArtifact("org.jetbrains.kotlin:kotlin-stdlib-jdk8:" + configuration.getKotlin()), null));
+            builder.addLibrary(resolver);
+        }
+        // Jellyfish end - Kotlin library downloader
+
         PaperPluginClassLoader classLoader = builder.buildClassLoader(jul, source, file, configuration);
         return new PaperPluginParent(source, file, configuration, classLoader, context);
     }
