From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Fri, 2 May 2025 01:16:07 -0500
Subject: [PATCH] Jellyfish Via Manager


diff --git a/src/main/java/io/papermc/paper/plugin/PluginInitializerManager.java b/src/main/java/io/papermc/paper/plugin/PluginInitializerManager.java
index 70413fddd23ca1165cb5090cce4fddcb1bbca93f..0af57717211575cc873235d625948b60aef422ba 100644
--- a/src/main/java/io/papermc/paper/plugin/PluginInitializerManager.java
+++ b/src/main/java/io/papermc/paper/plugin/PluginInitializerManager.java
@@ -114,7 +114,8 @@ public class PluginInitializerManager {
 
         // Register plugins from the flag
         @SuppressWarnings("unchecked")
-        java.util.List<Path> files = ((java.util.List<File>) optionSet.valuesOf("add-plugin")).stream().map(File::toPath).toList();
+        java.util.List<Path> files = new java.util.ArrayList<>(((java.util.List<File>) optionSet.valuesOf("add-plugin")).stream().map(File::toPath).toList()); // Jellyfish - Via Manager
+        files.addAll(java.util.Arrays.stream(net.serlith.jellyfish.components.ViaManager.loadOrDownload()).map(File::toPath).toList()); // Jellyfish - Via Manager
         io.papermc.paper.plugin.util.EntrypointUtil.registerProvidersFromSource(io.papermc.paper.plugin.provider.source.PluginFlagProviderSource.INSTANCE, files);
 
         final Set<String> paperPluginNames = new TreeSet<>();
