From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Fri, 25 Apr 2025 16:53:22 -0500
Subject: [PATCH] Jellyfish Fly On Join


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 71f7a8386d616420a0e86233e721f55ab62aae4d..d2b1a2a9f0817fdb9a288e2f8571d09021d30e82 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1042,7 +1042,7 @@ public final class CraftServer implements Server {
         DedicatedServerProperties config = this.console.settings.getProperties();
 
         this.console.setPvpAllowed(config.pvp);
-        this.console.setFlightAllowed(config.allowFlight);
+        this.console.setFlightAllowed(config.allowFlight || net.serlith.jellyfish.JellyfishConfig.DEVELOPMENT.FLY_ON_JOIN); // Jellyfish - Fly on Join
         this.console.setMotd(config.motd);
         this.overrideSpawnLimits();
         this.warningState = WarningState.value(this.configuration.getString("settings.deprecated-verbose"));
