--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2334,6 +_,20 @@
         }
         // Paper end
 
+        // Jellyfish start - Config files setup
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getJellyfishConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        @NotNull
+        public java.util.Properties getServerProperties()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Jellyfish end - Config files setup
+
         /**
          * Sends the component to the player
          *
