--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1088,6 +_,7 @@
 
         org.spigotmc.SpigotConfig.init((File) this.console.options.valueOf("spigot-settings")); // Spigot
         this.console.paperConfigurations.reloadConfigs(this.console);
+        net.serlith.jellyfish.JellyfishConfig.init((File) console.options.valueOf("jellyfish-settings")); // Jellyfish - Config files setup
         for (ServerLevel world : this.console.getAllLevels()) {
             // world.serverLevelData.setDifficulty(config.difficulty); // Paper - per level difficulty
             world.setSpawnSettings(world.serverLevelData.getDifficulty() != Difficulty.PEACEFUL && config.spawnMonsters); // Paper - per level difficulty (from MinecraftServer#setDifficulty(ServerLevel, Difficulty, boolean))
@@ -1120,6 +_,7 @@
         org.spigotmc.SpigotConfig.registerCommands(); // Spigot
         io.papermc.paper.command.PaperCommands.registerCommands(this.console); // Paper
         this.spark.registerCommandBeforePlugins(this); // Paper - spark
+        net.serlith.jellyfish.JellyfishConfig.registerCommands(); // Jellyfish - Config files setup
         this.overrideAllCommandBlockCommands = this.commandsConfiguration.getStringList("command-block-overrides").contains("*");
         this.ignoreVanillaPermissions = this.commandsConfiguration.getBoolean("ignore-vanilla-permissions");
 
@@ -3050,6 +_,20 @@
         {
             return CraftServer.this.console.paperConfigurations.createLegacyObject(CraftServer.this.console);
         }
+
+        // Jellyfish start - Config files setup
+        @Override
+        public YamlConfiguration getJellyfishConfig()
+        {
+            return net.serlith.jellyfish.JellyfishConfig.config;
+        }
+
+        @Override
+        public java.util.Properties getServerProperties()
+        {
+            return getProperties().properties;
+        }
+        // Jellyfish end - Config files setup
 
         @Override
         public void restart() {
